plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -0,0 +1,657 @@
+
+		$shortcut_meta_query = array();
+		foreach ( array( 'key', 'value', 'compare', 'type', 'compare_key', 'type_key' ) as $key ) {
+			if ( $this->arg_isset( "meta_{$key}" ) ) {
+				$shortcut_meta_query[ $key ] = $this->args[ "meta_{$key}" ];
+			}
+		}
+
+		if ( ! empty( $shortcut_meta_query ) ) {
+			if ( ! empty( $this->args['meta_query'] ) ) {
+				$this->args['meta_query'] = array( // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+					'relation' => 'AND',
+					$shortcut_meta_query,
+					$this->args['meta_query'],
+				);
+			} else {
+				$this->args['meta_query'] = array( $shortcut_meta_query ); // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+			}
+		}
+	}
+
+	/**
+	 * Sanitizes the 'status' query var.
+	 *
+	 * @return void
+	 */
+	private function sanitize_status(): void {
+		// Sanitize status.
+		$valid_statuses = array_keys( wc_get_order_statuses() );
+
+		if ( empty( $this->args['status'] ) || 'any' === $this->args['status'] ) {
+			$this->args['status'] = $valid_statuses;
+		} elseif ( 'all' === $this->args['status'] ) {
+			$this->args['status'] = array();
+		} else {
+			$this->args['status'] = is_array( $this->args['status'] ) ? $this->args['status'] : array( $this->args['status'] );
+
+			foreach ( $this->args['status'] as &$status ) {
+				$status = in_array( 'wc-' . $status, $valid_statuses, true ) ? 'wc-' . $status : $status;
+			}
+
+			$this->args['status'] = array_unique( array_filter( $this->args['status'] ) );
+		}
+	}
+
+	/**
+	 * Parses and sanitizes the 'orderby' query var.
+	 *
+	 * @return void
+	 */
+	private function sanitize_orderby(): void {
+		// Allowed keys.
+		// TODO: rand, meta keys, etc.
+		$allowed_keys = array( 'ID', 'id', 'type', 'date', 'modified', 'parent' );
+
+		// Translate $orderby to a valid field.
+		$mapping = array(
+			'ID'       => "{$this->tables['orders']}.id",
+			'id'       => "{$this->tables['orders']}.id",
+			'type'     => "{$this->tables['orders']}.type",
+			'date'     => "{$this->tables['orders']}.date_created_gmt",
+			'modified' => "{$this->tables['orders']}.date_updated_gmt",
+			'parent'   => "{$this->tables['orders']}.parent_order_id",
+		);
+
+		$order   = $this->args['order'] ?? '';
+		$orderby = $this->args['orderby'] ?? '';
+
+		if ( 'none' === $orderby ) {
+			return;
+		}
+
+		if ( is_string( $orderby ) ) {
+			$orderby = array( $orderby => $order );
+		}
+
+		$this->args['orderby'] = array();
+		foreach ( $orderby as $order_key => $order ) {
+			if ( isset( $mapping[ $order_key ] ) ) {
+				$this->args['orderby'][ $mapping[ $order_key ] ] = $this->sanitize_order( $order );
+			}
+		}
+	}
+
+	/**
+	 * Makes sure the order in an ORDER BY statement is either 'ASC' o 'DESC'.
+	 *
+	 * @param string $order The unsanitized order.
+	 * @return string The sanitized order.
+	 */
+	private function sanitize_order( string $order ): string {
+		$order = strtoupper( is_string( $order ) ? $order : '' );
```

Comment:
the `is_string` check is not required, `$order` is being coerced to string implicitly because of type declaration